Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm

```diff
@@ -965,6 +983,9 @@ - (BOOL)prefersStatusBarHidden
         else
             removeClientForContext(secondContextId);
     }
+
+    if (ensureModel(secondContextId)->isChildOfElementFullscreen())
+        videosInElementFullscreenChanged();
```

Comment:
I don't think this is necessary; this method is about native fullscreen, not Element fullscreen, and swapping native fullscreen states won't effect whether the video element has a fullscreen ancestor.